File: GeneratedEntityReaderClasses/Components/PhysicsJoint2Component.cs
namespace GeneratedNoitaClasses;
public class PhysicsJoint2Component : NoitaComponentBase,  iNoitaType<PhysicsJoint2Component>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUShort joint_id { get; set; } = new();
	public NoitaFloat break_force { get; set; } = new();
	public NoitaFloat break_distance { get; set; } = new();
	public NoitaBool break_on_body_modified { get; set; } = new();
	public NoitaFloat break_on_shear_angle_deg { get; set; } = new();
	public NoitaUInt type { get; set; } = new();
	public NoitaInt body1_id { get; set; } = new();
	public NoitaInt body2_id { get; set; } = new();
	public NoitaFloat offset_x { get; set; } = new();
	public NoitaFloat offset_y { get; set; } = new();
	public NoitaFloat ray_x { get; set; } = new();
	public NoitaFloat ray_y { get; set; } = new();
	public NoitaFloat surface_attachment_offset_x { get; set; } = new();
	public NoitaFloat surface_attachment_offset_y { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		joint_id.Read(s);
		break_force.Read(s);
		break_distance.Read(s);
		break_on_body_modified.Read(s);
		break_on_shear_angle_deg.Read(s);
		type.Read(s);
		body1_id.Read(s);
		body2_id.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);
		ray_x.Read(s);
		ray_y.Read(s);
		surface_attachment_offset_x.Read(s);
		surface_attachment_offset_y.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		joint_id.Write(s);
		break_force.Write(s);
		break_distance.Write(s);
		break_on_body_modified.Write(s);
		break_on_shear_angle_deg.Write(s);
		type.Write(s);
		body1_id.Write(s);
		body2_id.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);
		ray_x.Write(s);
		ray_y.Write(s);
		surface_attachment_offset_x.Write(s);
		surface_attachment_offset_y.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsJoint2Component.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUShort
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class PhysicsJoint2Component:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    joint_id:NoitaUShort = field(default_factory=NoitaUShort)
    break_force:NoitaFloat = field(default_factory=NoitaFloat)
    break_distance:NoitaFloat = field(default_factory=NoitaFloat)
    break_on_body_modified:NoitaBool = field(default_factory=NoitaBool)
    break_on_shear_angle_deg:NoitaFloat = field(default_factory=NoitaFloat)
    type:NoitaUInt = field(default_factory=NoitaUInt)
    body1_id:NoitaInt = field(default_factory=NoitaInt)
    body2_id:NoitaInt = field(default_factory=NoitaInt)
    offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    ray_x:NoitaFloat = field(default_factory=NoitaFloat)
    ray_y:NoitaFloat = field(default_factory=NoitaFloat)
    surface_attachment_offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    surface_attachment_offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.joint_id.Read(s)
        self.break_force.Read(s)
        self.break_distance.Read(s)
        self.break_on_body_modified.Read(s)
        self.break_on_shear_angle_deg.Read(s)
        self.type.Read(s)
        self.body1_id.Read(s)
        self.body2_id.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
        self.ray_x.Read(s)
        self.ray_y.Read(s)
        self.surface_attachment_offset_x.Read(s)
        self.surface_attachment_offset_y.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.joint_id.Write(s)
        self.break_force.Write(s)
        self.break_distance.Write(s)
        self.break_on_body_modified.Write(s)
        self.break_on_shear_angle_deg.Write(s)
        self.type.Write(s)
        self.body1_id.Write(s)
        self.body2_id.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)
        self.ray_x.Write(s)
        self.ray_y.Write(s)
        self.surface_attachment_offset_x.Write(s)
        self.surface_attachment_offset_y.Write(s)

